File: SDKs/botify.com/1.0.0/python/src/sdk/models/shared/csvexportstatus.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import urlsquery as shared_urlsquery


@dataclass_json
@dataclasses.dataclass
class CsvExportStatus:
    query: shared_urlsquery.UrlsQuery = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('query') }})
    area: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('area') }})
    date_created: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('date_created'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    job_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('job_id') }})
    job_status: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('job_status') }})
    job_url: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('job_url') }})
    nb_results: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('nb_results') }})
    results: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('results') }})
    

File: SDKs/botify.com/1.0.0/typescript/src/sdk/models/shared/csvexportstatus.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { UrlsQuery } from "./urlsquery";



export class CsvExportStatus extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=area" })
  area?: number;

  @SpeakeasyMetadata({ data: "json, name=date_created" })
  dateCreated?: Date;

  @SpeakeasyMetadata({ data: "json, name=job_id" })
  jobId?: string;

  @SpeakeasyMetadata({ data: "json, name=job_status" })
  jobStatus?: string;

  @SpeakeasyMetadata({ data: "json, name=job_url" })
  jobUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=nb_results" })
  nbResults?: number;

  @SpeakeasyMetadata({ data: "json, name=query" })
  query: UrlsQuery;

  @SpeakeasyMetadata({ data: "json, name=results" })
  results?: string;
}
